sdk/core/System.Net.ClientModel/src/ResultOfT.cs

```diff
@@ -7,12 +7,18 @@
 
 namespace System.Net.ClientModel;
 
-public class Result<T> : NullableResult<T>
+public class Result<T> : Result
 {
-    internal Result(T value, MessageResponse response) : base(value, response)
+    private T _value;
+    private MessageResponse _response;
+
+    internal Result(T value, MessageResponse response)// : base(value, response)
     {
-        Debug.Assert(value != null);
-        Debug.Assert(response != null);
+        _value = value;
+        _response = response!;
+
+        // TODO: note this will throw in the current implementation
+        // And we do want to keep this validation for correctness of this type.
```

Comment:
@christothes, I think this is why the inheritance hierarch shouldn't go in this direction.

It's still an open question, I think, if we want to preserve the inheritance hierarchy.